add release note for Composer 2.8.3 (#9345)
pantheon-systems/documentation: source/releasenotes/2024-12-10-composer-2-8-3.md

---
title: Composer 2.8.3 now available
published_date: "2024-12-10"
categories: [infrastructure,tools-apis]
---

Composer 2.8.3 is now available on Pantheon as of December 10, 2024.

### No action required

> Composer 2.8.3 will slowly roll out to all Integrated Composer sites on Pantheon automatically. There is no action required on your part.

Reviewer: I think Composer is globally installed on all sites, not just those with IC.

Author: Yes but I don't think you would see or interact with it if you weren't using IC.

Reviewer: In fact we should clarify if this change is to
- runtimes
- Integrated Composer build containers
- Both?

> 

Reviewer: ```suggestion
Composer commands can be run against sites on Pantheon via a [Terminus plugin](https://github.com/pantheon-systems/terminus-composer-plugin). For operations that modifying version controlled code (like adding a module via `composer require` which will change `composer.json` and `composer.lock`) you must run such commands in Dev or Multidev environments where version controlled files are writable. Such files are not writeable in Test and Live environments.

```

Author: @stevector 

When would I use Terminus to `composer require` a thing rather than `composer require`ing locally and pushing the updated `composer.json`? Is that process different?

Author: Also: is this note relevant to the 2.8.3 _update_ to Composer specifically? I don't know that we should be adding this level of detail outside the documentation for Composer on the platform unless we're going to copy/paste this with every update to Composer. I think linking to the Composer docs should be sufficient.

---
title: Composer 2.8.3 now available
published_date: "2024-12-10"
categories: [infrastructure,tools-apis]
---

Composer 2.8.3 is now available on Pantheon as of December 10, 2024.

### Automatic rollout - No action required

> Composer 2.8.3 will gradually roll out to all Integrated Composer sites on Pantheon automatically. No action required from you during this process.

Reviewer: Even sites without IC have Composer _available_ as an executable. I don't want to overexplain though.

---
title: Composer 2.8.3 now available
published_date: "2024-12-10"
categories: [infrastructure,tools-apis]
---

Composer 2.8.3 is now available on Pantheon as of December 10, 2024.

### Automatic rollout - No action required

Composer 2.8.3 will gradually roll out to automatically. No action required from you during this process.

> Composer commands can be run against sites on Pantheon via the [Terminus Composer plugin](https://github.com/pantheon-systems/terminus-composer-plugin). For operations that modify version controlled code (like adding a module via `composer require` which will change `composer.json` and `composer.lock`) you must run such commands in Dev or Multidev environments where version controlled files are writable. Such files are not writeable in Test and Live environments.

Author: ```suggestion

Composer commands can be run against sites on Pantheon via the [Terminus Composer plugin](https://github.com/pantheon-systems/terminus-composer-plugin). For operations that modify version controlled code (like adding a module via `composer require` which will change `composer.json` and `composer.lock`) you must run such commands in Dev or Multidev environments where version controlled files are writable. Such files are not writeable in Test and Live environments.
```

---
title: Composer 2.8.3 now available
published_date: "2024-12-10"
categories: [infrastructure,tools-apis]
---

Composer 2.8.3 is now available on Pantheon as of December 10, 2024.

### Automatic rollout - No action required

Composer 2.8.3 will gradually roll out to automatically. No action required from you during this process.

> 
> Composer commands can be run against sites on Pantheon via the [Terminus Composer plugin](https://github.com/pantheon-systems/terminus-composer-plugin). For operations that modify version controlled code (like adding a module via `composer require` which will change `composer.json` and `composer.lock`) you must run such commands in Dev or Multidev environments where version controlled files are writable. Such files are not writeable in Test and Live environments.

Author: ```suggestion
```

### Highlights

Composer 2.8.3 introduces a range of improvements and updates made between versions 2.7.7 and 2.8.3, including:

* **`--patch-only` flag for updates**: Restrict updates to patch versions with this new flag for the `update` command.
* **Improved interactive package updates**: Enhanced package display makes interactive updates more intuitive.
* **JSON schema validation**: New validation ensures integrity for `composer.lock` files.
* **PHP 8.4 compatibility**: Compatibility updates for the latest PHP version.
* **Bug fixes**: Resolved issues with provider suggestions and lock file schemas.

View an overview of the changes made between [2.7.7 and 2.8.3 by comparing the tags](https://github.com/composer/composer/compare/2.7.7...2.8.3) or explore the highlights on [Composer's Releases page](https://github.com/composer/composer/releases).

> 

Author: ```suggestion

### Notes

Composer commands can be run against sites on Pantheon via the [Terminus Composer plugin](https://github.com/pantheon-systems/terminus-composer-plugin). For operations that modify version controlled code (like adding a module via `composer require` which will change `composer.json` and `composer.lock`) you must run such commands in Dev or Multidev environments where version controlled files are writable. Such files are not writeable in Test and Live environments.
```

---
title: Composer 2.8.3 now available

> published_date: "2024-12-10"

Reviewer 2: ```suggestion
published_date: "2025-01-16"
```

categories: [infrastructure,tools-apis]
---

> Composer 2.8.3 is now available on Pantheon as of December 10, 2024.

Reviewer 2: now available makes me think i have to opt in
```suggestion
Composer 2.8.3 is now used in both runtime environments and integrated composer build environments on Pantheon.
```

### Automatic rollout - No action required

> Composer 2.8.3 will gradually roll out to automatically. No action required from you during this process.

Reviewer 2: ```suggestion
Composer 2.8.3 will roll out to automatically. No action required from you during this process.
```

* **Improved interactive package updates**: Enhanced package display makes interactive updates more intuitive.
* **JSON schema validation**: New validation ensures integrity for `composer.lock` files.
* **PHP 8.4 compatibility**: Compatibility updates for the latest PHP version.
* **Bug fixes**: Resolved issues with provider suggestions and lock file schemas.

View an overview of the changes made between [2.7.7 and 2.8.3 by comparing the tags](https://github.com/composer/composer/compare/2.7.7...2.8.3) or explore the highlights on [Composer's Releases page](https://github.com/composer/composer/releases).

### Notes

Composer commands can be run against sites on Pantheon via the [Terminus Composer plugin](https://github.com/pantheon-systems/terminus-composer-plugin). For operations that modify version controlled code (like adding a module via `composer require` which will change `composer.json` and `composer.lock`) you must run such commands in Dev or Multidev environments where version controlled files are writable. Such files are not writeable in Test and Live environments.

> To learn more about Composer on Pantheon, visit our guide on [Integrated Composer](https://docs.pantheon.io/guides/integrated-composer).

Reviewer 2: ```suggestion
To learn more about Composer on Pantheon, visit our guide on [Integrated Composer](/guides/integrated-composer).
```
